### PR TITLE
feat(jira): add batch_create_versions

### PR DESCRIPTION
**Pull request by Otoris**
## Description

Implements Milestone 3: Batch Create Jira Project Versions

Adds the `batch_create_versions` tool and backend logic to allow creating multiple fix versions in a single operation for a specified Jira project. This enables efficient, atomic version management.

## Changes

- Adds `batch_create_versions` tool for Jira (MCP API and backend)
- Accepts `project_key` and a JSON array of version objects (`name`, `startDate`, `releaseDate`, `description`)
- Iterates over the batch, calling single-version creation logic for each entry
- Returns an array of result objects, each indicating success or error for the corresponding version
- Handles edge cases: empty input, partial failures, and invalid JSON
- Documents partial success strategy and error reporting
- Updates and adds unit tests for all-success, partial-failure, all-failure, and empty batch scenarios

## Testing

- [x] Unit tests added/updated for all batch creation scenarios and error cases
- [x] All tests pass locally (`pytest`)
- [x] Manual check: Successfully created multiple versions in a test Jira project

## Checklist

- [x] Code follows project style guidelines (linting passes, pre-commit clean)
- [x] Tests added/updated for changes
- [x] All tests pass locally
- [x] Documentation updated (README and tool doc)

### CODE REVIEW

File: tests/unit/servers/test_jira_server.py [hunk 1]
```diff
@@ -298,6 +299,10 @@ async def test_lifespan(app: FastMCP) -> AsyncGenerator[MainAppContext, None]:
     jira_sub_mcp.tool()(remove_issue_link)
     jira_sub_mcp.tool()(transition_issue)
     jira_sub_mcp.tool()(update_sprint)
+    jira_sub_mcp.tool(name="jira_batch_create_versions")(
+        batch_create_versions
+    )  # <-- ensure correct tool name
+    jira_sub_mcp.tool(name="batch_create_versions")(batch_create_versions)
```
Comment (sooperset) [inline tests/unit/servers/test_jira_server.py]: The registration of the `batch_create_versions` tool appears to be duplicated

**Review by sooperset — COMMENTED** [review]
Thanks for the contribution @Otoris! I've rebased the branch and updated the README.md. 

There's one minor issue to address before merging: `batch_create_versions` registration appears to be duplicated in the tests. Could you please remove the duplicate registration? Once that's fixed, we'll be ready to merge.

**Comment by Otoris** [comment]
Fixed test duplicate - ready for re-review

**Review by sooperset — APPROVED** [review]
Great work, @Otoris! thanks for fixing the duplicate registration issue.
